Add Intent API implementation that proxies to underlying kube-apiserver (#46005)
grafana/grafana: docs/sources/administration/configuration.md

### enable_custom_baselayers

Set this to `true` to disable loading other custom base maps and hide them in the Grafana UI. Default is `false`.

## [intentapi]

Settings related to the Intent API.

## [intentapi.server]

Settings related to the Intent API HTTP server.

### listen_address

Which address the Intent API will be listening on, must be in `host:port` format. Default value is `127.0.0.1:8443`.

### cert_file_path

> Path to TLS certificate that would be used by the Intent API. Since k8s effectively enforces TLS, this is required. Unset by default, make sure to set it if using the Intent API.

Reviewer: ```suggestion
Path to TLS certificate that would be used by the Intent API. Since k8s effectively enforces TLS, this is required. Unset by default, it must be set to enable Intent API.
```

### key_file_path

Path to TLS key file that would be used by the Intent API. Since k8s effectively enforces TLS, this is required. Unset by default, make sure to set it if using the Intent API.

### read_timeout

The maximum duration for reading the entire request, including the body. A zero or negative value means there will be no timeout. Default value is `1m`.

### write_timeout

The maximum duration before timing out writes of the response. It is reset whenever a new request's header is read. A zero or negative value means there will be no timeout. Default value is `1m`.

### shutdown_timeout

The maximum duration before timing out graceful server shutdown. Default value is `1m`.

> ## [intentapi.proxy]

Reviewer: It's not a blocker, especially because i don't have a better alternative offhand, but i'm a little dubious about using the word "proxy" here.

i think of a proxy as:
1. A relatively dumb intermediary
2. Comprising both the input and output ends of that intermediary

Here, we're using "proxy" to refer to configuring just the "output" end. And while our _implementation_ is currently proxy-like in its dumb-ness, that may change in the future, even if we never go the route of possibly embedding the apiserver into the Grafana backend.

Author: Do you think it'll be OK to change the naming later on? I would expect so, as long as we stabilise it before making this a stable feature.

If so, then I'll leave the naming problem for later, once we have a better understanding of what the backend looks like.

WDYT?
